infra/gcp/lib_util.sh

```diff
@@ -67,3 +67,19 @@ fi
 function indent() {
     ${SED} -u 's/^/  /'
 }
+
+# Join things with separator
+# Arguments:
+#   $1:  Separator
+#   $2+: The things to join
+# Example usage:
+#   join_by , foo bar baz
+function join_by() {
+  if [ $# -lt 2 ] || [ -z "${1}" ] || [ -z "${2}" ]; then
+      echo "join_by(separator, string...) requires at least 2 arguments" >&2
+      return 1
+  fi
+
+  local IFS="${1}"; shift
+  echo "$*"
```

Comment:
To ensure someone doesn't "fix" this, please comment on why this is $* rather than $@ ?  $@ doesn't seem to respect IFS.

Reply:
Done